Repository/ProductRepository.cs

```diff
@@ -0,0 +1,71 @@
+﻿using EFCoreServices.Models;
+using Microsoft.EntityFrameworkCore;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace EFCoreServices.Repository
+{
+    public class ProductRepository : IProductRepository
+    {
+        ProductDBContext db;
+        public ProductRepository(ProductDBContext _DbContext)
+        {
+            db = _DbContext;
+        }
+
+        public async Task<int> AddProduct(Product prod)
+        {
+            if(db != null)
+            {
+                await db.Products.AddAsync(prod);
+                await db.SaveChangesAsync();
+
+            }
+            return 0;
+        }
+
+        public async Task<int> DeleteProduct(int? ProdID)
+        {
+            int result = 0;
+            if(db != null)
+            {
+                var prod = await db.Products.FirstOrDefaultAsync(e => e.Id == ProdID);
+
+                if(prod != null)
```

Comment:
if its not found the end user should know, hence use single which will throw an exception and handle it in your service and controller layers to send back a not found